meta-lmp-base/recipes-sota/fioefi/fioefi/fioefi.sh.in

```diff
@@ -0,0 +1,58 @@
+#!/bin/sh
+#
+# Copyright (C) 2024 Foundries.IO
+#
+# SPDX-License-Identifier: MIT
+#
+set -e
+
+scriptName="$(basename "$0")"
+
+error_exit() {
+	if [ -n "${1}" ] ; then
+		echo "ERROR:  ${1}"
+	else
+		echo "ERROR: Command execution failure"
+	fi
+	exit 1
+}
+
+debug() {
+	if [ -n "${1}" ] ; then
+		echo "DEBUG: ${1}"
+	else
+		echo "DEBUG: Command execution failure"
+	fi
+}
+
+matches() {
+	input="$1"
+	pattern="$2"
+	echo "$input" | grep -q "$pattern"
+}
+
+@@INCLUDE_SOC_FUNCTIONS@@
+
+# Mock functions. On real targets only *_soc implementations
+# should be used
+getenv() {
+	debug "Get value of '${1}'"
+}
+
+setenv() {
+	debug "Assign value '${1} = ${2}'"
+}
+
+if matches "$scriptName" "printenv"; then
+	if type 'getenv_soc' 2>/dev/null | grep -q 'function'; then
+ 		getenv_soc "${1}"
+	else
+		getenv "${1}"
+	fi
+elif matches "$scriptName" "setenv"; then
+	if type 'setenv_soc' 2>/dev/null | grep -q 'function'; then
+ 		setenv_soc "${1}" "${2}"
+	else
+		setenv "${1}" "${2}"
+	fi
+fi
```

Comment:
delenv is missing. Is this intentional?

Reply:
I plan to add it also, however I'm still not sure if it's feasible to implement a proper "state machine" here. The main purpose of this script was to keep the same API as we have now for fiofb/ubootenv to avoid intrusive changes in ak-lite, but not to emulate a full functionality of fiovb/ubootenv

Do you somehow delete variables (`fiovb_delenv <var>`) in CI tests?

Reply:
no. Maybe I should have such a test for a custom variable.